### PR TITLE
Add operation to query lock status.

### PR DESCRIPTION
**Pull request by tanbangcheng**

### CODE REVIEW

**Comment by johnou** [comment]
What's the use case? Given the lock state may change after isLocked has returned I would recommend tryLock then something..

**Comment by tanbangcheng** [comment]
@johnou Just check the state but not lock it, like Redisson's RLock.  
use RLock to mark whether a switch is managed by the Controller (then the Controller locks, and do something),  if not locked, the manager assigns the lock to one of the Controllers

**Comment by johnou** [comment]
@tanbangcheng rebase please?

File: core/src/main/java/io/atomix/core/lock/impl/DistributedLockService.java [hunk 1]
```diff
@@ -49,4 +49,11 @@ default void lock(int lockId) {
   @Command("unlock")
   void unlock(int lockId);
 
+  /**
+   * Query whether the lock state.
+   *
+   * @return {@code true} if this lock is locked, {@code false} otherwise
+   */
+  @Operation(value = "isLocked", type = OperationType.QUERY)
```
Comment (johnou) [inline core/src/main/java/io/atomix/core/lock/impl/DistributedLockService.java]: @tanbangcheng query annotation?

**Comment by johnou** [comment]
@tanbangcheng rebase?

**Pull request merged**